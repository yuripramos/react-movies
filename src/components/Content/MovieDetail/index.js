import MovieDetail from "./MovieDetail";

import { connect } from "redux-zero/react";

import contentActions from "../../../actions/Movies";

export default connect(
  ({ filteringResultsByRating, displayMovieDetails, isMovieDetailAction }) => ({
    filteringResultsByRating,
    displayMovieDetails,
    isMovieDetailAction
  }),
  contentActions
)(MovieDetail);
