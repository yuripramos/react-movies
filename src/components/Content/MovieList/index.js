import MovieList from "./MovieList";

import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import SearchActions from "../../../actions/Search";
import contentActions from "../../../actions/Movies";

export default connect(
  ({ displayMovieDetails, ClearFilterOnSearch }) => ({
    displayMovieDetails,
    ClearFilterOnSearch
  }),
  combineActions(contentActions, SearchActions)
)(MovieList);
