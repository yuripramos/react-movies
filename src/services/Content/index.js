import axios from "axios";
import { API_URL } from "../../utils/constants";

export async function getAuthors() {
  return axios
    .get(`${API_URL}/5be5e3ae2f00005b000fc3f6`)
    .then(resp => resp.data);
}

export async function getArticles() {
  return axios
    .get(`${API_URL}/5be5e3fa2f000082000fc3f8`)
    .then(resp => resp.data);
}
