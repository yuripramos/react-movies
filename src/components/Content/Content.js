import React, { Component, Fragment } from "react";

import { Container } from "../../styles/grid";
import MovieList from "./MovieList";
import MovieDetail from "./MovieDetail";
import { arrayOf, object, func } from "prop-types";
import Hide from "../common/Hide";
import { isResponsive } from "../../utils/getResolution";

import { ContentWrapper, Column } from "./styles";

class Content extends Component {
  constructor(props) {
    super(props);
    this.innerRef = React.createRef();
  }

  componentDidMount() {
    const { getArticles, getAuthors } = this.props;
    getAuthors();
    getArticles();
  }

  render() {
    const { articlesList, authorsList } = this.props;
    const isFilled = articlesList && articlesList.length > 0;
    return (
      <Container>
        <ContentWrapper>
          {isFilled && (
            <Fragment>
              <Column width={isResponsive() ? 0 : 30}>
                <Hide below="lg">
                  <MovieList
                    authorsList={authorsList}
                    articlesList={articlesList}
                  />
                </Hide>
              </Column>
              <Column width={isResponsive() ? 100 : 70}>
                <MovieDetail
                  authorsList={authorsList}
                  articlesList={articlesList}
                  innerRef={this.innerRef}
                />
              </Column>
            </Fragment>
          )}
        </ContentWrapper>
      </Container>
    );
  }
}

Content.defaultProps = {};

Content.propTypes = {
  getArticles: func,
  getAuthors: func,
  articlesList: arrayOf(object),
  authorsList: arrayOf(object)
};

export default Content;
