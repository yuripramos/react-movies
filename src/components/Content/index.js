import Content from "./Content";

import { connect } from "redux-zero/react";

import contentActions from "../../actions/Articles";

export default connect(
  ({ articlesList, authorsList, getAuthors, getArticles }) => ({
    articlesList,
    authorsList,
    getAuthors,
    getArticles
  }),
  contentActions
)(Content);
