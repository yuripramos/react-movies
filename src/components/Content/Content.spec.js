import React from "react";
import { shallow } from "enzyme";
import Content from "./Content";
import ArticleDetail from "./ArticleDetail";

const props = {
  getAuthors: jest.fn(),
  getArticles: jest.fn(),
  authorsList: [
    {
      name: "Jemma Chadwick",
      id: 1
    },
    {
      name: "Nicholas Jordan",
      id: 2
    }
  ],
  articlesList: [
    {
      title: "AssCo has revamped the theory of versioning",
      body: "We will mesh the buzzword. Our feature set is unparallele",
      metadata: {
        publishedAt: 1492004832000,
        authorId: 2
      }
    },
    {
      title: "Matleh",
      body: "she is no take arms and to take arms.",
      metadata: {
        publishedAt: 1431006432000,
        authorId: 1
      }
    }
  ]
};

describe("Content", () => {
  it("should match snapshot", () => {
    expect(shallow(<Content {...props} />)).toMatchSnapshot();
  });
  it("should render ArticlesDetail", () => {
    expect(shallow(<Content {...props} />).find(ArticleDetail).length).toBe(1);
  });
});
