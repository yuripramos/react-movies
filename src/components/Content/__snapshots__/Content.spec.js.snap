// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should match snapshot 1`] = `
<Container
  blacklist={
    Array [
      "css",
      "width",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "fontSize",
      "color",
      "bg",
      "flex",
      "order",
      "alignSelf",
    ]
  }
>
  <styled.div>
    <styled.div
      width={0}
    >
      <Hide
        above={null}
        below="lg"
      >
        <ArticleList
          articlesList={
            Array [
              Object {
                "body": "We will mesh the buzzword. Our feature set is unparallele",
                "metadata": Object {
                  "authorId": 2,
                  "publishedAt": 1492004832000,
                },
                "title": "AssCo has revamped the theory of versioning",
              },
              Object {
                "body": "she is no take arms and to take arms.",
                "metadata": Object {
                  "authorId": 1,
                  "publishedAt": 1431006432000,
                },
                "title": "Matleh",
              },
            ]
          }
          authorsList={
            Array [
              Object {
                "id": 1,
                "name": "Jemma Chadwick",
              },
              Object {
                "id": 2,
                "name": "Nicholas Jordan",
              },
            ]
          }
        />
      </Hide>
    </styled.div>
    <styled.div
      width={100}
    >
      <ArticleDetail
        articlesList={
          Array [
            Object {
              "body": "We will mesh the buzzword. Our feature set is unparallele",
              "metadata": Object {
                "authorId": 2,
                "publishedAt": 1492004832000,
              },
              "title": "AssCo has revamped the theory of versioning",
            },
            Object {
              "body": "she is no take arms and to take arms.",
              "metadata": Object {
                "authorId": 1,
                "publishedAt": 1431006432000,
              },
              "title": "Matleh",
            },
          ]
        }
        authorsList={
          Array [
            Object {
              "id": 1,
              "name": "Jemma Chadwick",
            },
            Object {
              "id": 2,
              "name": "Nicholas Jordan",
            },
          ]
        }
        innerRef={
          Object {
            "current": null,
          }
        }
      />
    </styled.div>
  </styled.div>
</Container>
`;
